10/jdk/alpine/slim-java.sh

```diff
@@ -136,56 +134,56 @@ function charset_files() {
 
 	# 2.3 Special treat for removing ZOS specific charsets
 	echo -n "INFO: Trimming charsets..."
-	mkdir -p ${root}/charsets_class
-	pushd ${root}/charsets_class >/dev/null
-		jar -xf ${root}/jre/lib/charsets.jar
+	mkdir -p "${root}"/charsets_class
+	pushd "${root}"/charsets_class >/dev/null || return
+		jar -xf "${root}"/jre/lib/charsets.jar
 		ibmEbcdic="290 300 833 834 838 918 930 933 935 937 939 1025 1026 1046 1047 1097 1112 1122 1123 1364"
 
 		# Generate sfj-excludes-charsets list as well. (OpenJ9 expects the file to be named sfj-excludes-charsets).
-		[ ! -e ${root}/jre/lib/slim/sun/nio/cs/ext/sfj-excludes-charsets ] || rm -rf ${root}/jre/lib/sfj/sun/nio/cs/ext/sfj-excludes-charsets
+		[ ! -e "${root}"/jre/lib/slim/sun/nio/cs/ext/sfj-excludes-charsets ] || rm -rf "${root}"/jre/lib/sfj/sun/nio/cs/ext/sfj-excludes-charsets
 		exclude_charsets=""
 
 		for charset in ${ibmEbcdic};
 		do
-			rm -f sun/nio/cs/ext/IBM${charset}.class
-			rm -f sun/nio/cs/ext/IBM${charset}\$*.class
+			rm -f sun/nio/cs/ext/IBM"${charset}".class
+			rm -f sun/nio/cs/ext/IBM"${charset}"\$*.class
 
 			exclude_charsets="${exclude_charsets} IBM${charset}"
 		done
-		mkdir -p $root/jre/lib/slim/sun/nio/cs/ext
-		echo ${exclude_charsets} > ${root}/jre/lib/slim/sun/nio/cs/ext/sfj-excludes-charsets
-		cp ${root}/jre/lib/slim/sun/nio/cs/ext/sfj-excludes-charsets sun/nio/cs/ext/
+		mkdir -p "$root"/jre/lib/slim/sun/nio/cs/ext
+		echo "${exclude_charsets}" > "${root}"/jre/lib/slim/sun/nio/cs/ext/sfj-excludes-charsets
+		cp "${root}"/jre/lib/slim/sun/nio/cs/ext/sfj-excludes-charsets sun/nio/cs/ext/
 
-		jar -cfm ${root}/jre/lib/charsets.jar META-INF/MANIFEST.MF *
-	popd >/dev/null
-	rm -rf ${root}/charsets_class
+		jar -cfm "${root}"/jre/lib/charsets.jar META-INF/MANIFEST.MF ./*
+	popd >/dev/null || return
+	rm -rf "${root}"/charsets_class
 	echo "done"
 }
 
 # Trim the rt.jar classes. The classes deleted are as per slim-java_rtjar_del.list
 function rt_jar_classes() {
 	# 2.4 Remove classes in rt.jar
 	echo -n "INFO: Trimming classes in rt.jar..."
-	mkdir -p ${root}/rt_class
-	pushd ${root}/rt_class >/dev/null
-		jar -xf ${root}/jre/lib/rt.jar
-		mkdir -p ${root}/rt_keep_class
-		for class in $(cat ${keep_list} | grep -v "^#");
+	mkdir -p "${root}"/rt_class
+	pushd" ${root}"/rt_class >/dev/null || return
```

Comment:
not sure what is better : `|| return` or `|| exit`. @dinogun any suggestions ?

Reply:
Is that `"` in the wrong place?  I think return is OK